src/v/cluster/topic_table.cc

```diff
@@ -356,9 +353,6 @@ topic_table::apply(create_non_replicable_topic_cmd cmd, model::offset o) {
 
     auto ca = tp->second.configuration;
     ca.cfg.tp_ns = new_non_rep_topic;
-    for (auto& assignment : ca.assignments) {
-        assignment.group = raft::group_id(-1);
-    }
```

Comment:
why is this not set to -1 for the non replicable topics?

Reply:
To prevent additional lookups when creating a materialized topic in `create_non_replicable_partition`. Also to me it doesn't seem to stretch the semantics too far given that all properties of a materialized partition are just copies of its parent.